### PR TITLE
Merging opensearch-dsl-py to opensearch-py

### LINKED ISSUE #194: [PROPOSAL] Merge opensearch-dsl-py into this repository
### What are you proposing?
Merge the opensearch-dsl-py repo here as an additional package. Users would be able to import both the dsl libraries and the python-client since dsl functions always require importing the python client.

### How did you come up with this proposal?
Currently the [ruby client](https://github.com/opensearch-project/opensearch-ruby) has multiple gems that are published independently of each other while living in the same repository. 

### What is the user experience going to be?
The user experience would be unchanged from how it is today, however it will make developer experience much better. Maintainers will have to release one less client as both the dsl libraries and the python client can be released together. 

### Why should it be built? Any reason not to?
Can't think of any. 

### What will it take to execute?
Porting over the client here should be fairly straightforward. It is unclear at this time what it takes to merge the source here and how new releases will happen.

### What are remaining open questions?
- Will this make discovery of the dsl client harder?
- Is this even feasible? Happy to be proven wrong if not.

### PR DESCRIPTION
**Pull request by saimedhi**
Signed-off-by: saimedhi <saimedhi@amazon.com>

### Description
Merging opensearch-dsl-py to opensearch-py

### Issues Resolved
Closes #194  

By submitting this pull request, I confirm that my contribution is made under the terms of the Apache 2.0 license.
For more information on following Developer Certificate of Origin and signing off your commits, please check [here](https://github.com/opensearch-project/OpenSearch/blob/main/CONTRIBUTING.md#developer-certificate-of-origin).

### CODE REVIEW

File: USER_GUIDE.md [hunk 1]
```diff
@@ -228,6 +230,33 @@ print('\n The deleted point in time:')
 print(response)
 ```
 
+## Using High-level Python client
```
Comment (harshavamsi) [inline USER_GUIDE.md:233]: Is this the only piece that is being carried forward from the high level client?

**Review by harshavamsi — CHANGES_REQUESTED** [review]
Can you also update the documentation to include the new DSL helpers in the `docs` folder?

File: USER_GUIDE.md (hunk 1 above)
Comment (saimedhi) [reply USER_GUIDE.md:233]: All other API's added have similar implementation. So, I added only Search API example.

**Review by harshavamsi — APPROVED** [review]

**Comment by adilnaimi** [comment]
When will this PR be released?

**Comment by VachaShah** [comment]
> When will this PR be released?

@harshavamsi Can you comment here?

**Comment by harshavamsi** [comment]
> > When will this PR be released?
> 
> @harshavamsi Can you comment here? Tracking in #307

Released here -- https://github.com/opensearch-project/opensearch-py/releases/tag/v2.2.0

**Pull request merged**